src/compiler/expression/assignment.rs

```diff
@@ -771,6 +771,7 @@ mod test {
             result
                 .program
                 .final_type_state()
+                .state
```

Comment:
Calling `.state` on something called `_state()` feels off, any thoughts on a rename to make the result of `.final_type_state()` clearer? `.final_type_info()` matching the return maybe?